lib/net/networking_engine.js

```diff
@@ -360,8 +366,17 @@ shaka.net.NetworkingEngine = class extends shaka.util.FakeEventTarget {
           // copying buffers if this is a partial view.
           request.body = shaka.util.BufferUtils.toArrayBuffer(request.body);
         }
-        return requestFilter(type, request, advType);
+        return requestFilter(type, request, context);
       });
+    };
+
+    if (this.onRequest_) {
```

Comment:
My concern is that we should add this filter only if the user enables CMCD, otherwise it should not be logged. On some older platforms, every filter we add makes the app slower. If you have not detected that error, I agree to approve this PR.

Reply:
The `onRequest` handle was written to be CMCD neutral. It could be used to apply additional non-removable filter logic in the future. I've updated the code in `player.js` to only pass the handler when CMCD is enabled.